Makefile

```
@@ -0,0 +1,79 @@
+XARGS_CMD ?= xargs -I {}
+BIN_DIR ?= ${HOME}/bin
+PATH := $(BIN_DIR):${PATH}
+
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+
+.PHONY: %/install %/lint release/%
+
+guard/program/%:
+	@ which $* > /dev/null || $(MAKE) $*/install
+
+$(BIN_DIR):
+				@ echo "[make]: Creating directory '$@'..."
+				mkdir -p $@
+
+zip/install:
+				@ echo "[$@]: Installing $(@D)..."
+				apt-get install zip -y
+				@ echo "[$@]: Completed successfully!"
+
+terraform/install: TERRAFORM_VERSION ?= $(shell curl -sSL https://checkpoint-api.hashicorp.com/v1/check/terraform | jq -r -M '.current_version')
+terraform/install: TERRAFORM_URL ?= https://releases.hashicorp.com/terraform/$(TERRAFORM_VERSION)/terraform_$(TERRAFORM_VERSION)_linux_amd64.zip
+terraform/install: | $(BIN_DIR)
+				@ echo "[$@]: Installing $(@D)..."
+				@ echo "[$@]: TERRAFORM_URL=$(TERRAFORM_URL)"
+				curl -sSL -o terraform.zip "$(TERRAFORM_URL)"
```

Might as well be consistent and use `$(@D)` everywhere...
```suggestion
				curl -sSL -o $(@D).zip "$(TERRAFORM_URL)"
```